cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -265,7 +320,7 @@ func buildDNSNameString(labels ...string) string {
 			res = fmt.Sprintf("%s.%s", label, res)
 		}
 	}
-	return res
+	return strings.ToLower(res)
```

Comment:
I would prefer the ToLower in the place(s) where we might get a non-lower-case string only.